mobile/lib/modules/asset_viewer/views/gallery_viewer.dart

```diff
@@ -373,6 +373,26 @@ class GalleryViewerPage extends HookConsumerWidget {
       );
     }
 
+    ImageProvider imageProvider(Asset asset) {
+      if (asset.isLocal) {
+        return localImageProvider(asset);
+      } else {
+        if (isLoadOriginal.value) {
+          return originalImageProvider(asset);
+        } else if (isLoadPreview.value) {
+          return remoteThumbnailImageProvider(
+            asset,
+            api.ThumbnailFormat.JPEG,
+          );
+        } else {
+          return remoteThumbnailImageProvider(
+            asset,
+            api.ThumbnailFormat.WEBP,
+          );
+        }
+      }
+    }
+
```

Comment:
Can you refactor this by removing the else-branches?
They seem redundant here and too much nesting is always hard to read.

Reply:
Thanks for pointing this out. It sort of ballooned on me. So much that I hardly remembered this being in the changes for this PR! I've removed the messy nesting and added a few comments.